crates/install-wheel-rs/src/wheel.rs

```diff
@@ -797,29 +798,29 @@ pub fn read_record_file(record: &mut impl Read) -> Result<Vec<RecordEntry>, Erro
         .collect()
 }
 
-/// Parse a file with `Key: value` entries such as WHEEL and METADATA
-fn parse_key_value_file(
+/// Parse a file with email message format such as WHEEL and METADATA
+fn parse_email_message_file(
     file: impl Read,
     debug_filename: &str,
 ) -> Result<FxHashMap<String, Vec<String>>, Error> {
     let mut data: FxHashMap<String, Vec<String>> = FxHashMap::default();
 
     let file = BufReader::new(file);
-    for (line_no, line) in file.lines().enumerate() {
-        let line = line?.trim().to_string();
-        if line.is_empty() {
-            continue;
-        }
-        let (key, value) = line.split_once(':').ok_or_else(|| {
-            Error::InvalidWheel(format!(
-                "Line {} of the {debug_filename} file is invalid",
-                line_no + 1
-            ))
-        })?;
-        data.entry(key.trim().to_string())
+    let content = file.bytes().collect::<Result<Vec<u8>, _>>()?;
+
+    let (headers, _) = parse_headers(content.as_slice()).map_err(|err| {
+        Error::InvalidWheel(format!("Failed to parse {debug_filename} file: {err}"))
+    })?;
+
+    for header in headers {
+        let name = header.get_key();
+        let value = header.get_value();
+
+        data.entry(name.trim().to_string())
```

Comment:
I thought we could maybe avoid these trims now that we're using `mailparse` but looks like whitespace is retained for the keys at least. We could probably avoid _allocating_ here though if there isn't leading / trailing whitespace. Micro-opt for the future.

Reply:
Yes, the name could be avoided.
Should we keep it for the values?
I would say it would prevent badly formatted values like we say in the ziglang package

Reply:
By the way, contrary to `mail-parser`, the `get_key()` function is returning a String -> useless `to_string()` below

Reply:
I think we should keep the trim functionality, but we can maybe avoid the extra `to_string` if there's nothing to trim.

Reply:
In other words: lets keep `trim` for both. But we _could_ skip `.to_string` if `name == name.trim()`, to avoid the allocation. But maybe more trouble than it's worth, I defer to you :)

Reply:
Updated! Do you think I should extract the trim logic to an other function?